core-bundle/src/Image/Studio/Studio.php

```diff
@@ -67,13 +68,13 @@ public function createLightboxImage($filePathOrImage, string $url = null, $sizeC
     public static function getSubscribedServices(): array
     {
         return [
-            self::class,
+            'contao.image.studio' => self::class,
             'contao.image.picture_factory' => PictureFactoryInterface::class,
             'contao.image.image_factory' => ImageFactoryInterface::class,
             'contao.image.resizer' => ResizerInterface::class,
             'contao.assets.files_context' => ContaoContext::class,
             'contao.framework' => ContaoFramework::class,
-            'event_dispatcher' => 'event_dispatcher',
+            'event_dispatcher' => EventDispatcherInterface::class,
```

Comment:
Using the class names would again make use of autowiring, right?
So we would have to use only the service names I think:
```suggestion
            'contao.image.studio',
            'contao.image.picture_factory',
            'contao.image.image_factory',
            'contao.image.resizer',
            'contao.assets.files_context',
            'contao.framework',
            'event_dispatcher',
```

Reply:
> Using the class names would again make use of autowiring, right?

I'd say so, too. The key is only used for the internal reference - you could for instance set `'foo' => 'event_dispatcher'` and then use it like `$container->get('foo')`. Writing `<numeric-key> => <value>` is equivalent to `<value> => <value>`.

Reply:
If I change this, however, I get the following error:

```
Symfony\Component\DependencyInjection\Exception\InvalidArgumentException: "Contao\CoreBundle\Image\Studio\Studio::getSubscribedServices()" must return valid PHP types for service "contao.image.studio" key "0", "contao.image.studio" returned.
```

Reply:
It would work like this, however, I am not sure if this is better:

```php
return [
    'contao.image.studio' => 'contao.image.studio',
    'contao.image.picture_factory' => 'contao.image.picture_factory',
    'contao.image.image_factory' => 'contao.image.image_factory',
    'contao.image.resizer' => 'contao.image.resizer',
    'contao.assets.files_context' => 'contao.assets.files_context',
    'contao.framework' => 'contao.framework',
    'event_dispatcher' => 'event_dispatcher',
];
```

Reply:
Ah maybe the shortcut is only working with class names. It sure is better if we're not using the FQCN version, so that you're able to decorate the service.

Reply:
Fixe in 58a480e8dac91ed3a7cb93bea6eeaa4206d7b8f3

Reply:
Changed in 58a480e8dac91ed3a7cb93bea6eeaa4206d7b8f3.